apis/core/v1beta1/alertingcluster_types.go

```diff
@@ -0,0 +1,74 @@
+package v1beta1
+
+import (
+	opnimeta "github.com/rancher/opni/pkg/util/meta"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AlertingDeployConfStandalone = "standalone"
+	AlertingDeployConfCluster    = "ha"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type AlertingCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              AlertingClusterSpec   `json:"spec,omitempty"`
+	Status            AlertingClusterStatus `json:"status,omitempty"`
+}
+
+type AlertingClusterSpec struct {
+	//+kubebuilder:validation:required
+	Gateway      corev1.LocalObjectReference `json:"gateway,omitempty"`
+	Alertmanager AlertManagerSpec            `json:"alertmanager,omitempty"`
+}
+
+type AlertManagerSpec struct {
+	Enable     bool                `json:"enable,omitempty"`
+	Image      *opnimeta.ImageSpec `json:"image,omitempty"`
+	LogLevel   string              `json:"logLevel,omitempty"`
+	DeployConf string
+
+	// Overrides for out-of-the box alerting specs
+	ApplicationSpec AlertingApplicationSpec `json:"workloads,omitempty"`
+}
+
+type AlertingApplicationSpec struct {
+	Replicas             *int32                            `json:"replicas,omitempty"`
+	ExtraArgs            []string                          `json:"extraArgs,omitempty"`
+	ExtraVolumes         []corev1.Volume                   `json:"extraVolumes,omitempty"`
+	ExtraVolumeMounts    []corev1.VolumeMount              `json:"extraVolumeMounts,omitempty"`
```

Comment:
Not required for this iteration but a lot of operators combine the extra Volume and Volume mounts into a single object as a QoL feature because most of the time these are 1:1